### PR TITLE
[ONNX] Update 'person_of_interest.rst', 'CODEOWNERS' and 'merge_rules.yaml'

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126364

The following are all constrained under the ONNX exporter project scope.

- `personal_of_interest.rst`
  - Moving folks no longer working on the project to emeritus.
  - Adding @justinchuby, @titaiwangms, @shubhambhokare1 and @xadupre,
    who have all made countless contributions to this project.
- `CODEOWNERS`
  - Removing folks no longer working on the project.
  - Updating new owners who will now be notified with PRs related to
    the specific file paths.
- `merge_rules.yaml`
  - Removing folks no longer working on the project.
 
🫡 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Review by titaiwangms — APPROVED** [review]

File: docs/source/community/persons_of_interest.rst [hunk 1]
```diff
@@ -226,9 +226,13 @@ C10 utils and operator dispatch
 
 ONNX exporter
 ~~~~~~~~~~~~~
--  Aaron Bockover (`abock <https://github.com/abock>`__)
--  Bowen Bao (`BowenBao <https://github.com/BowenBao>`__)
--  Thiago Crepaldi (`thiagocrepaldi <https://github.com/thiagocrepaldi>`__)
+-  Shubham Bhokare (`shubhambhokare1 <https://github.com/shubhambhokare1>`__)
```
Comment (justinchuby) [inline docs/source/community/persons_of_interest.rst:229]: Add Rama too?

**Comment by titaiwangms** [comment]
@malfet PTAL

**Review by justinchuby — APPROVED** [review]

**Review by albanD — APPROVED** [review]
Sounds good!

**Pull request closed without merging**